iroh-bytes/src/protocol.rs

```diff
@@ -517,6 +521,81 @@ impl GetRequest {
     }
 }
 
+/// A query request
+///
+/// A query request is identical to a get request, except that it describes a
+/// set of chunks for which we want availability information.
+///
+/// The answer to a query request is a set of chunks that are available, encoded
+/// as a [`RangeSpecSeq`].
+///
+/// The provider has to compute the availability information only for the requested
+/// ranges, not for the entire blob or collection. The response must contain
+/// availability information for the requested ranges, but may contain availability
+/// information for additional ranges if expedient. E.g. if the provider has
+/// a blob completely available, it may send `RangeSet2::all()` for that blob
+/// instead of limiting the response to the requested ranges.
```

Comment:
Do we think the size of a response would get too big if we always respond with all the availability for the entire blob/collection?  It would make the protocol a lot simpler.

Reply:
The issue is that depending on the impl, computing availability can be quite expensive. Either you go straight to the bytes, then computing availability is just hashing the file again. Or you cache it, then it might be incorrect.

That is the rationale for specifying a range of interest, in addition to the response size.

Assuming you got the data cached, reducing the response to the range that the requester has asked for is a single op, so I don't think it is overly complex. We just need to specify more precisely what the response should be - it should be the smallest RangeSpec that describes the availability in the intervals that were asked for.

Reply:
sounds good